### ISSUE TITLE
Hangs after selecting template in Step 2

### ISSUE THREAD

**Issue by brucew2013** [veda-consulting/uk.co.vedaconsulting.mosaico#153]
OS: CentOS 7.3 x86-64 cPanel/WHM virtuozzo
Webserver: Litespeed 5.1.15
PHP: 7.1 
CMS: Joomla 3.7.0
CRM: CiviCRM 4.7.19

After needing to install Shoreditch and Flexmailer plus ImageMagick and FileInfo PHP modules it began to work.

In step 2, selecting the default template hangs and disables all (button) links in CiviCRM and Mosaico. (See attached; notice upper right corner "loading") and throwing JS error:
> Uncaught ReferenceError: $ is not defined at development.alcoholjustice.org/:42

Opening that up highlights line 42. Is this correct?:
```
41 <script type="text/javascript">
=>42    $(function() { 
43      if (!Mosaico.isCompatible()) {
44        alert('Update your browser!');
45        return;
46      }
```

<img width="733" alt="screen shot 2017-05-13 at 4 14 28 pm" src="https://cloud.githubusercontent.com/assets/5554148/26029301/6badcb6c-37f7-11e7-8172-38713cb7bc0a.png">

**Comment by brucew2013** [comment]
Additionally, in v.2, it looks like the path is incorrect. 
What browser console shows (see below) but doesn't exist in the Git ZIP package:
**JROOT/media/civicrm/ext/uk.co.vedaconsulting.mosaico/packages/mosaico/templates/versafix-1/edres/_full.png**

What exists is:
**JROOT/media/civicrm/ext/uk.co.vedaconsulting.mosaico/templates/CRM**

<img width="1289" alt="screen shot 2017-05-13 at 6 30 15 pm" src="https://cloud.githubusercontent.com/assets/5554148/26030019/8e66d906-380a-11e7-8f6a-f479ed7baef9.png">

**Comment by stevekessler** [comment]
This is also happening during step 1 if you are not using the Shoreditch.
<img width="1792" alt="screen shot 2017-05-14 at 10 17 25 pm" src="https://cloud.githubusercontent.com/assets/867985/26042534/816e0e1a-38f3-11e7-8210-e094d322ed37.png">

**Comment by rekaeps** [comment]
I'm having the same issue in Drupal 7.54 and CiviCRM 4.7.19. Running PHP 7.

**Comment by smch** [comment]
Same problem, Civi 4.7.20

**Comment by totten** [comment]
Background: 
 * Mosaico is a significant JS application in its own right, and `uk.co.vedaconsulting.mosaico` embeds that application within CiviMail.
 * During installation, we need to obtain a copy of both the application and the extension. 
 * At runtime, the Mosaico app loads in an IFRAME.  The app has its own full set of JS requirements (eg particular versions of jQuery and KnockoutJS). 

The current README points out a couple ways to install. These boil down to 2 options:

 * Get the zip file from `https://download.civicrm.org/extension/uk.co.vedaconsulting.mosaico`. This zip file contains all dependencies.
 * Get the code from Github and *then run `setup.sh` to download additional dependencies*. (This requires a fairly complete toolchain -- e.g. `git` and `npm`.)

Notably, if you try to wing it (e.g. just download+install without following the docs), then the system will be missing a bunch of dependencies. After reproducing this (mis)installation, I was able to produce a screenshot much like @brucew2013's. 

(Aside: Big thanks for including the screenshots!)

Possibly interesting to note... this particular issue hasn't been reported in recent months. It may be that the docs explain the installation better now, or it may be that the first symptom has changed, or that people figure they don't need to comment any more. But IMHO, the general mistake is still possible.

What we should probably do... include a system status check which generates a better error if you're missing the dependencies.

**Comment by totten** [comment]
I've merged #217 which displays a more astute warning if the `packages` are missing. Thanks to everyone who reported.

If the problem persists somehow, feel free to comment again.
